code/missionui/missiondebrief.cpp

```diff
@@ -720,6 +721,8 @@ void debrief_multi_server_stuff()
 		stages[i] = stage_active[i];
 		for (j=0; j<debriefp->num_stages; j++) {
 			if ( eval_sexp(debriefp->stages[j].formula) ) {
+				// DISCUSSME: how to handle repeatedly evaluating SEXPs that have side effects?
```

Comment:
IMO, we shouldn't do anything special; it's up to the mission designer to make sure their SEXP won't break by being evaluated for each team.

Reply:
Ok, but since in TVT missions, a stage's SEXP is evaluated multiple times, at what point do we perform text replacement on the stage? The first time the stage's SEXP evaluates to true?

Reply:
For a TvT it seems sensible not to try to second guess what the FREDder might intend. A FREDder might want different teams to have different values displayed. For instance, a "You need $points-required to win" message should be different for team 1 and 2.

Reply:
@Karajorma And that's exactly the issue I was concerned about. I don't think there's a way to do this.

The only debrief info the server sends to clients is which debriefing stages to display (by index); see `send_debrief_info()` in multimsgs.cpp. The stage indices for each team are selected in this function we're talking about.

In fact, AFAICT there is no way for the server to send updated debriefing text to clients, so I'm not sure text replacement can even work in multi.

Are mission variables and variable operations (e.g., modify) replicated on the clients? If so, then we *might* have a chance of making text replacement work client-side, but I'm not sure there's a way to perform the sort of context-sensitive text replacement with the side effects you expect, or at least, not without a major rewrite of the `DEBRIEF_INFO` packet definition. But that'd be way out of scope.

Reply:
I pushed the best possible solution for multi I can think of. Please see the list of gotchas in the PR's OP.

Reply:
On second thought, I wonder why  I haven't seen this issue with briefings in multi. Maybe I overlooked something in the code....

Reply:
Won't the debriefing stage SEXPs be evaluated on the client?  For TvT missions, each client will belong to a particular team, so the stage SEXP evaluation should necessarily take place on the client, not the server.  And therefore, each client will have its own copy of the text, which *should* have client-specific text replacement.

This is all educated guesswork though.  It will be necessary to run some TvT checks to see if this is a correct summary of stage handling.

Reply:
SEXPs are never evaluated on the client. Only ever on the server. Text replacement however will be done on the client side. 

Anything that is labelled as a network variable should automatically be sent to the clients. The problem then is that the server will evaluate the event for both teams and then either send out two packets or send one containing the last value. This means you're going to need two variables for the TvT example I gave above. Nothing that can be done about that.

Reply:
Ah. so variables marked as "Network" are replicated on the clients.

And yeah, per your TvT example, surely you can't have a single variable `points` that has different values for team 1 and team 2. How would the server maintain an authoritative value for `points` in that case?

So, to sum up the **multi-only** gotchas:
1. Text replacement works only for network variables.
2. SEXPs in debriefings should never have side effects.
3. Text replacement in debriefings, however, can have side effects.

Are these ok? Any other concerns about the PR?

Note that, as a consequence, unless we support "Network Containers", containers will not be usable for text replacement in multi.

Reply:
No other concerns from me.  Just make sure the "gotchas" are documented somewhere that modders will see them.